pkgs/build-support/kernel/closure.rs

```diff
@@ -0,0 +1,184 @@
+use std::os::unix::ffi::{OsStrExt,OsStringExt};
+use std::ffi::{OsStr,OsString};
+use std::path::{Path,PathBuf};
+
+use std::env;
+use std::fs;
+use std::process::{Command,exit};
+use std::io::Write;
+use std::io::ErrorKind;
+
+use std::collections::HashSet;
+use std::collections::VecDeque;
+use std::hash::Hash;
+
+fn main() {
+    let args : Vec<OsString> = env::args_os().collect();
+    let allow_missing = env::var_os("allowMissing").is_some();
+    let firmware = env::var_os("firmware").unwrap();
+    let data = ModulesData {
+        kernel: Path::new(&args[1]).to_path_buf(),
+        out_dir: Path::new(&args[2]).to_path_buf(),
```

Comment:
why pass some through as env vars and some as argv, instead of using envvars for everything?

Reply:
Especially since everything after argv[3] is a root

Reply:
good point